src/NextCloud/NextCloudVideoEdit.jsx

```diff
@@ -142,64 +155,71 @@ class Edit extends Component {
       >
-        {data.url ? (
-          <Body data={this.props.data} isEditMode={true} />
+        {data.url && this.isValidUrl(data.url) ? (
+          <Body data={data} isEditMode={true} />
         ) : (
-          <Message>
-            <center>
-              <img src={videoBlockSVG} alt="" />
-              <div className="toolbar-inner">
-                <Input
-                  onKeyDown={this.onKeyDownVariantMenuForm}
-                  onChange={this.onChangeUrl}
-                  placeholder={placeholder}
-                  value={this.state.url}
-                  // Prevents propagation to the Dropzone and the opening
-                  // of the upload browser dialog
-                  onClick={(e) => e.stopPropagation()}
-                />
-                {this.state.url && (
-                  <Button.Group>
-                    <Button
-                      basic
-                      className="cancel"
-                      onClick={(e) => {
-                        e.stopPropagation();
-                        this.setState({ url: '' });
-                      }}
-                    >
-                      <Icon name={clearSVG} size="30px" />
-                    </Button>
-                  </Button.Group>
-                )}
+          <center>
+            <img src={videoBlockSVG} alt="" />
+            <div className="toolbar-inner">
+              <Input
+                onKeyDown={this.onKeyDownVariantMenuForm}
+                onChange={this.onChangeUrl}
+                placeholder={placeholder}
+                value={this.state.url}
+                // Prevents propagation to the Dropzone and the opening
+                // of the upload browser dialog
+                onClick={(e) => e.stopPropagation()}
+                error={data.url && !this.isValidUrl(data.url)}
+              />
+              {this.state.url && (
                 <Button.Group>
                   <Button
                     basic
-                    primary
+                    className="cancel"
                     onClick={(e) => {
                       e.stopPropagation();
-                      this.onSubmitUrl();
+                      this.setState({ url: '' });
                     }}
                   >
-                    <Icon name={aheadSVG} size="30px" />
+                    <Icon name={clearSVG} size="30px" />
                   </Button>
                 </Button.Group>
-              </div>
-            </center>
-          </Message>
+              )}
+              <Button.Group>
+                <Button
+                  basic
+                  primary
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    this.onSubmitUrl();
+                  }}
+                >
+                  <Icon name={aheadSVG} size="30px" />
+                </Button>
+              </Button.Group>
+            </div>
+            {data.url && !this.isValidUrl(data.url) && (
```

Comment:
@rexalex If the video URL is not valid, it shouldn't save it with block when you press Document Save. Remove the the broken URL, in order to make sure we don't have a broken video block in View mode. See attached video:

[youtube.webm](https://user-images.githubusercontent.com/323385/221984344-59c0725c-bef1-4b78-89b0-a1e1fd82953a.webm)